Content.Server/_RMC14/Rules/CMDistressSignalRuleSystem.cs

```diff
@@ -908,18 +932,53 @@ private Spawners GetSpawners()
         return spawners;
     }
 
-    private (EntProtoId Id, EntityUid Ent) NextSquad(ProtoId<JobPrototype> job, CMDistressSignalRuleComponent rule)
+    private Dictionary<string, int> GetSquadJobNumber(ProtoId<JobPrototype> job, CMDistressSignalRuleComponent rule)
     {
-        // TODO RMC14 this biases people towards alpha as that's the first one, maybe not a problem once people can pick a preferred squad?
-        if (!rule.NextSquad.TryGetValue(job, out var next) ||
-            next >= rule.SquadIds.Count)
+        var jobNumPerSquad = new Dictionary<string, int>();
+
+        var members = EntityQueryEnumerator<SquadMemberComponent>();
+
+        foreach (var squad in rule.SquadIds)
         {
-            rule.NextSquad[job] = 0;
-            next = 0;
+            jobNumPerSquad[squad] = 0;
         }
 
-        var id = rule.SquadIds[next++];
-        rule.NextSquad[job] = next;
+        while (members.MoveNext(out var uid, out var member))
+        {
+            var name = member.Squad == null ? null : Name(member.Squad.Value);
+            var squadName = "Squad" + name;
+            _jobs.MindTryGetJob(_mind.GetMind(uid), out _, out var prototype);
+            if (prototype! == job)
+            {
+                jobNumPerSquad[squadName]++;
+            }
```

Comment:
This should be using rule.SquadIds like abov or EntityUid not concatenating the prototype name of the squad like this.